cmd/util/cmd/extract-payloads-by-address/cmd.go

```diff
@@ -0,0 +1,262 @@
+package extractpayloads
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/fxamacker/cbor/v2"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+
+	"github.com/onflow/cadence/runtime/common"
+
+	"github.com/onflow/flow-go/ledger"
+)
+
+const (
+	defaultBufioWriteSize = 1024 * 32
+	defaultBufioReadSize  = 1024 * 32
+
+	payloadEncodingVersion = 1
+)
+
+var (
+	flagInputPayloadFileName  string
+	flagOutputPayloadFileName string
+	flagAddresses             string
+)
+
+var Cmd = &cobra.Command{
+	Use:   "extract-payload-by-address",
+	Short: "Read payload file and generate payload file containing payloads with specified addresses",
+	Run:   run,
+}
+
+func init() {
+	Cmd.Flags().StringVar(
+		&flagInputPayloadFileName,
+		"input-filename",
+		"",
+		"Input payload file name")
+	_ = Cmd.MarkFlagRequired("input-filename")
+
+	Cmd.Flags().StringVar(
+		&flagOutputPayloadFileName,
+		"output-filename",
+		"",
+		"Output payload file name")
+	_ = Cmd.MarkFlagRequired("output-filename")
+
+	Cmd.Flags().StringVar(
+		&flagAddresses,
+		"addresses",
+		"",
+		"extract payloads of addresses (comma separated hex-encoded addresses) to file specified by output-payload-filename",
+	)
+	_ = Cmd.MarkFlagRequired("addresses")
+}
+
+func run(*cobra.Command, []string) {
+
+	if _, err := os.Stat(flagInputPayloadFileName); os.IsNotExist(err) {
+		log.Fatal().Msgf("Input file %s doesn't exist", flagInputPayloadFileName)
+	}
+
+	if _, err := os.Stat(flagOutputPayloadFileName); os.IsExist(err) {
+		log.Fatal().Msgf("Output file %s exists", flagOutputPayloadFileName)
+	}
+
+	addresses, err := parseAddresses(strings.Split(flagAddresses, ","))
+	if err != nil {
+		log.Fatal().Err(err)
+	}
+
+	log.Info().Msgf(
+		"extracting payloads with address %v from %s to %s",
+		addresses,
+		flagInputPayloadFileName,
+		flagOutputPayloadFileName,
+	)
+
+	numOfPayloadWritten, err := extractPayloads(log.Logger, flagInputPayloadFileName, flagOutputPayloadFileName, addresses)
+	if err != nil {
+		log.Fatal().Err(err)
+	}
+
+	err = overwritePayloadCountInFile(flagOutputPayloadFileName, numOfPayloadWritten)
+	if err != nil {
+		log.Fatal().Err(err)
+	}
+}
+
+func overwritePayloadCountInFile(output string, numOfPayloadWritten int) error {
+	in, err := os.OpenFile(output, os.O_RDWR, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to open %s to write payload count: %w", output, err)
+	}
+	defer in.Close()
+
+	var data [9]byte
+	data[0] = 0x1b
```

Comment:
What is this magic value?

Reply:
Good catch! This hex value is a CBOR integer indicator.

I refactored to use named variable in [75e1099](https://github.com/onflow/flow-go/pull/5389/commits/75e10997e1b8b74d4b6a648412bbc59c6d8a1479)